utils/yoke/src/cart.rs

```diff
@@ -0,0 +1,117 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{Yoke, Yokeable};
+use std::rc::Rc;
+use std::sync::Arc;
+
+/// A [`Cart`] is a type that is acceptable as backing storage in a [`Yoke`].
+///
+/// The essential invariant that must be maintained by implementors is that `Self::Inner` references
+/// obtained from this type via `Self::get_inner()` must continue to be valid for the
+/// duration of this type provided that this type is never accessed via `&mut` references.
+///
+/// For example, `Vec<u8>` is a valid [`Cart`], however `Vec<RefCell<u8>>` is not, because
+/// in the latter type it is possible to use interior mutation to change the data.
```

Comment:
I actually don't understand why this is true, and maybe indicates that Cart's contract is underspecified.

Reply:
Oh, I messed up the type. Fixing